tools/querytee/response_comparator.go

```diff
@@ -391,10 +430,17 @@ func compareVector(expectedRaw, actualRaw json.RawMessage, queryEvaluationTime t
 		return err
 	}
 
-	if allVectorSamplesWithinRecentSampleWindow(expected, queryEvaluationTime, opts) && allVectorSamplesWithinRecentSampleWindow(actual, queryEvaluationTime, opts) {
+	if allVectorSamplesOutsideComparableWindow(expected, queryEvaluationTime, opts) && allVectorSamplesOutsideComparableWindow(actual, queryEvaluationTime, opts) {
 		return nil
 	}
 
+	expected = filterSlice(expected, func(p *model.Sample) bool {
+		return int64(p.Timestamp) < opts.SkipSamplesBefore
+	})
+	actual = filterSlice(actual, func(p *model.Sample) bool {
+		return int64(p.Timestamp) < opts.SkipSamplesBefore
+	})
+
```

Comment:
Mutating the slices like this might lead to confusing error messsages later on. For example if `SkipSamplesBefore` is set to 1000, and the sample for a series has timestamp 1001 on one side and the corresponding sample on the other side has timestamp 999, then we'll return a `expected metric {...} missing from actual response` or `expected X metrics but got Y` error below but in reality we've filtered out one side.

A similar issue applies to the matrix comparison above as well.

Reply:
Giving the right error message will make this comparison a little more complex where we compare the filtered out part as well. I am not sure if I want to spend more time in making this part of error message better at the moment because I haven't heard of requiring this option other than block builder at the moment. Maybe put it as a TODO right now?

Reply:
Sure, I'm OK with adding a comment explaining that it might produce confusing error messages.

Reply:
I added a comment and also made the error message a little better - I added a note in the error message whenever we filtered them out.